spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/StreamDeploymentController.java

```diff
@@ -247,6 +255,7 @@ private void deployStream(StreamDefinition stream, Map<String, String> streamDep
 			isDownStreamAppPartitioned = isPartitionedConsumer(appDeployTimeProperties,
 					upstreamAppSupportsPartition);
 
+			logger.info(String.format("Downloading resource URI [%s]", registration.getUri()));
 			Resource appResource = registration.getResource();
 			Resource metadataResource = registration.getMetadataResource();
```

Comment:
Can we add a similar logging for the `metadataResource`as well with the explicit check on whether a separate `metadataResourceURI` exists.